Streetcode/Streetcode.XUnitTest/MediatRTests/News/DeleteNewsTest.cs

```diff
@@ -0,0 +1,93 @@
+﻿using Moq;
+using Streetcode.BLL.DTO.News;
+using Streetcode.BLL.MediatR.Newss.Delete;
+using Streetcode.DAL.Repositories.Interfaces.Base;
+using System.Linq.Expressions;
+using Xunit;
+
+namespace Streetcode.XUnitTest.MediatRTests.News
+{
+    public class DeleteNewsTest
+    {
+        private Mock<IRepositoryWrapper> _mockRepository;
+
+        public DeleteNewsTest()
+        {
+            _mockRepository = new Mock<IRepositoryWrapper>();
+        }
+
+        [Fact]
+        public async Task ShouldDeleteSuccessfully()
+        {
+            // Arrange
+            var testNews = GetNews();
+            SetupMockRepositoryGetFirstOrDefault(testNews);
+            _mockRepository.Setup(x => x.SaveChangesAsync()).ReturnsAsync(1);
+
+            var handler = new DeleteNewsHandler(_mockRepository.Object);
+
+            // Act
+            var result = await handler.Handle(new DeleteNewsCommand(testNews.Id), CancellationToken.None);
+
+            // Assert
+            Assert.Multiple(
+                () => Assert.NotNull(result),
+                () => Assert.True(result.IsSuccess)
+            );
+
+            _mockRepository.Verify(x => x.NewsRepository.Delete(It.Is<DAL.Entities.News.News>(x => x.Id == testNews.Id)), Times.Once);
+            _mockRepository.Verify(x => x.SaveChangesAsync(), Times.Once);
+        }
+
+        [Fact]
+        public async Task ShouldThrowException_IdNotExisting()
+        {
+            // Arrange
+            var testNews = GetNews();
+            var expectedError = $"No news found by entered Id - {testNews.Id}";
+            SetupMockRepositoryGetFirstOrDefault(null);
+
+            var handler = new DeleteNewsHandler(_mockRepository.Object);
+
+            // Act
+            var result = await handler.Handle(new DeleteNewsCommand(testNews.Id), CancellationToken.None);
+
+            // Assert
+            Assert.Equal(expectedError, result.Errors.First().Message);
+            _mockRepository.Verify(x => x.NewsRepository.Delete(It.IsAny<DAL.Entities.News.News>()), Times.Never);
+        }
+
+        [Fact]
+        public async Task ShouldThrowException_SaveChangesAsyncIsNotSuccessful()
+        {
+            // Arrange
+            var testNews = GetNews();
+            var expectedError = "Failed to delete news";
+            SetupMockRepositoryGetFirstOrDefault(testNews);
+            _mockRepository.Setup(x => x.SaveChanges()).Throws(new Exception(expectedError));
```

Comment:
you can set this value also in your mocking method passing the result of the method as a parameters